src/QsCompiler/Transformations/ClassicallyControlled.cs

```diff
@@ -160,6 +160,41 @@ public StatementTransformation(SyntaxTreeTransformation parent) : base(parent)
                     }
                 }
 
+                private (bool, QsConditionalStatement) ProcessNOT(QsConditionalStatement conditionStatement)
+                {
+                    // This method expects elif blocks to have been abstracted out
+                    if (conditionStatement.ConditionalBlocks.Length != 1)
+                    {
+                        return (false, conditionStatement);
+                    }
+
+                    var (condition, block) = conditionStatement.ConditionalBlocks[0];
+
+                    if (condition.Expression is ExpressionKind.NOT notCondition)
+                    {
+                        var newConditionalBlock = conditionStatement.Default;
+                        if (newConditionalBlock.IsNull)
+                        {
+                            var emptyScope = new QsScope(
+                                ImmutableArray<QsStatement>.Empty,
+                                LocalDeclarations.Empty);
+                            newConditionalBlock = QsNullable<QsPositionedBlock>.NewValue(
```

Comment:
There doesn't seem to be a need to wrap this in a QsNullable as it has an item and that item is what is used in line 189.

Reply:
Yup, you are right, I can restructure the shape of the if-statement and get rid of the extra layer here. I'll make that update!